Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -23,6 +24,7 @@ public class ProductStore: Store {
                 network: Network,
                 remote: ProductsRemoteProtocol,
                 generativeContentRemote: GenerativeContentRemoteProtocol) {
+        productVariationStorageManager = ProductVariationStorageManager(storageManager: storageManager)
```

Comment:
Just a design thought/question:  We're performing the different services for dealing with products as a `ProductStore` extension, however for variations here we're defining  its own `ProductVariationStorageManager` and encapsulated to its own class whatever behaviour we need to implement. 

It feels that this approach is better rather than dumping all methods into the `ProductStore`. Perhaps we should think of doing the same for Products, and leave `ProductStore`s responsibility just as a layer to connect the different services and dispatch the actions as needed without needing to know the details of what each service does.